docs/site/src/plugins/sui-docs-inject-code/lib/includesLoader.js

```diff
@@ -0,0 +1,174 @@
+"use strict";
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+const fs_1 = __importDefault(require("fs"));
+const path_1 = __importDefault(require("path"));
+const markdownLoader = function (source) {
+    let fileString = source;
+    const callback = this.async();
+    const options = this.getOptions();
+    const markdownFilename = path_1.default.basename(this.resourcePath);
+    const markdownFilepath = path_1.default.dirname(this.resourcePath);
+    const repoPath = path_1.default.join(__dirname, "../../../../../..");
+    // Do not load and render markdown files without docusaurus header.
+    // These files are only used to be included in other files and should not generate their own web page
+    if (fileString.length >= 3 && fileString.substring(0, 3) !== "---") {
+        return callback && callback(null, "");
+    }
+    function addMarkdownIncludes(fileContent) {
+        let res = fileContent;
+        const matches = fileContent.match(/\{@\w+: .+\}/g);
+        if (matches) {
+            matches.forEach((match) => {
+                const replacer = new RegExp(match, "g");
+                if (match.startsWith("{@inject: ")) {
+                    const injectFileFull = match.substring(10, match.length - 1);
+                    const injectFile = injectFileFull.substring(0, injectFileFull.indexOf("#") > 0
+                        ? injectFileFull.indexOf("#")
+                        : injectFileFull.length);
+                    let fileExt = injectFile.substring(injectFile.lastIndexOf(".") + 1);
+                    let language = "";
+                    const fullPath = path_1.default.join(repoPath, injectFile);
+                    switch (fileExt) {
+                        case "move":
+                            language = "rust";
+                            break;
+                        case "toml":
+                            language = "rust";
+                            break;
+                        case "lock":
+                            language = "rust";
+                            break;
```

Comment:
This seems a bit strange -- I can see an argument for borrowing Rust's syntax highlighter for Move, but TOML and lock files (which are also TOML files) look nothing like Rust.  Do we not have native support for them? If not, the closest thing we have is probably `ini` or `conf` files.

Reply:
Yeah, we have a move highlighter that I didn't know about at the time.